Platon/backend/app/follow_system/views.py

```diff
@@ -1,10 +1,250 @@
+            return make_response(jsonify(mylist), 200)
+
+        else:
+            return make_response(jsonify({'error': 'Input Format Error'}), 400)
+
+class GetFollowRequestsAPI(Resource):
+
+    @api.doc(responses={200: 'Follow Requests List is successfully returned',
+                        404: 'Follow Requests List is empty',
+                        400: 'Input Format Error',
+                        401: 'Current user is unauthorized to see the Follow Requests',
+                        500: 'Database Connection Error'})
+    @api.expect(get_follow_requests_parser)
+    @login_required
+    def post(user_id, self):
+        '''
+            Takes following_id as input and returns the follow requests as a list.
+        '''
+        form = GetFollowRequestsForm(request.form)
+        if form.validate():
+
+            # user_id should be equal to form.following_id.data
+            if user_id != form.following_id.data:
+                return make_response(jsonify({'error': 'Unauthorized'}), 401)
+
+            try:
+                followSearch = FollowRequests.query.filter(FollowRequests.following_id == form.following_id.data).all()
+            except:
+                return make_response(jsonify({'error': 'Database Connection Error'}), 500)
+
+            if followSearch is []:
+                return make_response(jsonify({'error': 'No Follow Request Found'}), 404)
+
+
+            mylist = []
+            for follow in followSearch:
+                mylist.append(follow.follower_id)
+
+            return make_response(jsonify(mylist), 200)
+
+        else:
+            return make_response(jsonify({'error': 'Input Format Error'}), 400)
+
+class SendFollowRequestAPI(Resource):
+
+    @api.doc(responses={200: 'Follow Request is sent successfully',
+                        400: 'Input Format Error',
+                        401: 'Current user is unauthorized to send Follow Request',
+                        500: 'Database Connection Error'})
+    @api.expect(send_follow_requests_parser)
+    @login_required
+    def post(user_id, self):
+        '''
+            Takes follower_id and following_id as inputs and creates the corresponding Follow Request.
+        '''
+        form = SendFollowRequestsForm(request.form)
+        if form.validate():
+
+            # Current user should be the follower.
+            if user_id != form.follower_id.data:
+                return make_response(jsonify({'error': 'Unauthorized'}), 401)
+
+            fr = FollowRequests(form.follower_id.data, form.following_id.data)
+            try:
+                db.session.add(fr)  # Creating a new database entry.
+                db.session.commit()
+            except:
+                return make_response(jsonify({'error': 'Database Connection Error'}), 500)
+
+            return make_response(jsonify({'msg': 'Follow Request is succesfully added'}), 200)
+
+        else:
+            return make_response(jsonify({'error': 'Input Format Error'}), 400)
```

Comment:
After you change previous endpoints to GET. You can also move this POST endpoints inside of these classes.

Reply:
Do you mean moving this post method into a class named e.g. FollowRequestsAPI where there will be 1 get method(getFollowRequests) and 2 post methods(accept or request)?

Reply:
You can not create 2 post methods for an endpoint. I think creating a follow request can be the post method, and answering the follow request can be the put or delete method. And also you can combine reject and accept follow request methods into a single method by adding an argument to the form.

But firstly, I think there is no need for a method that only creates a Follow Request. Because int our system, if account of a user is not private, you can follow immediately(no need to create Follow Request). So the only situation that is needed to create Follow Request is when a user tries to follow another user whose account is private. As a result, I think to perform this functionality you need to add an endpoint for this. If you add this endpoint then there will be no need to an endpoint which creates only Follow Request records.

Reply:
Okay, I got it. I am working on it. Thanks

Reply:
With the latest commit, public profiles can be followed directly.